packages/rsbuild-plugin-angular/src/lib/models/normalize-options.ts

```diff
@@ -1,44 +1,65 @@
-import { PluginAngularOptions } from './plugin-options';
-import { join, resolve } from 'path';
-import { existsSync } from 'fs';
+import { FileReplacement, PluginAngularOptions } from './plugin-options';
+import { join, resolve } from 'node:path';
+import { existsSync } from 'node:fs';
+
+/**
+ * Resolves file replacement paths to absolute paths based on the provided root directory.
+ *
+ * @param fileReplacements - Array of file replacements with relative paths.
+ * @param root - The root directory to resolve the paths against.
+ * @returns Array of file replacements resolved against the root.
+ */
+export function resolveFileReplacements(
+  fileReplacements: FileReplacement[],
+  root: string
+): FileReplacement[] {
+  return fileReplacements.map((fileReplacement) => ({
+    replace: resolve(root, fileReplacement.replace),
+    with: resolve(root, fileReplacement.with),
+  }));
+}
+
+export function getHasServer({
+  server,
+  ssrEntry,
+  root,
+}: Pick<PluginAngularOptions, 'server' | 'ssrEntry' | 'root'>): boolean {
+  return !!(
+    server &&
+    ssrEntry &&
+    existsSync(join(root, server)) &&
+    existsSync(join(root, ssrEntry))
+  );
+}
 
 export function normalizeOptions(
-  options: Partial<PluginAngularOptions>
+  options: Partial<PluginAngularOptions> = {}
```

Comment:
A normalize functions receives more open types, adds missing defaults and returns a normalized output. 
Therefore I opened the type here. It is non breaking and has no affect ti usages.